core/src/test/scala/org/graphframes/PatternMatchSuite.scala

```diff
@@ -547,6 +547,16 @@ class PatternMatchSuite extends SparkFunSuite with GraphFrameTestSparkContext {
     compareResultToExpected(res, Set(Row(0L, 1L, 2L), Row(2L, 0L, 1L), Row(1L, 2L, 0L)))
   }
 
```

Comment:
Can we get some more tests? Like a length 5? One just doesn't feel adequate.

Reply:
@rjurney Good point. Unsupported invalid queries too.

Reply:
Increasing the length, it is even more interesting to see how the same node is visited again and again within the cycle.